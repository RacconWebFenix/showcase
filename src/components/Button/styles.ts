import styled, { css } from 'styled-components'
import { ButtonTypes } from '.'

export const Wrapper = styled.button<ButtonTypes>`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    border: 0;
    cursor: pointer;
    color: ${theme.colors.mainBg};
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.large};
    text-decoration: none;
    line-height: ${theme.font.sizes.xlarge};
    display: inline-flex;
    align-items: center;
    justify-content: center;
    transition: all ease 0.2s;

    &:hover {
      background: ${theme.colors.hovered};
      box-shadow: 10px 10px 5px 0px #000000;
    }

    &:active {
      box-shadow: 5px 5px 10px 1px #000000;
    }
  `}
`
