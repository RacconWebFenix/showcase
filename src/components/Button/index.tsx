import * as S from './styles'

export type ButtonTypes = {
  size?: 'small' | 'medium' | 'large'
  children: React.ReactNode
  variant?: 'primary' | 'secondary'
}

const Button = ({ variant, children, size = 'medium' }: ButtonTypes) => (
  <S.Wrapper variant={variant} size={size}>
    {children}
  </S.Wrapper>
)

export default Button
