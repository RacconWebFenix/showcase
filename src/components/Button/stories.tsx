import { Meta, StoryObj } from '@storybook/react'
import { Share } from '@styled-icons/bootstrap/Share'

import Button from '.'

export default {
  title: 'Button',
  component: Button
} as Meta

export const Template: StoryObj = {
  args: {
    children: 'Ver Cases'
  }
}

export const SharedIcon: StoryObj = {
  args: {
    children: '',
    sharedicon: <Share />
  }
}
