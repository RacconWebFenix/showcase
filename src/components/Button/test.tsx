import { screen } from '@testing-library/react'

import Button from '.'
import { renderWithTheme } from '../../utils/tests/helpers'

describe('<Button />', () => {
  it('should render a button and with variant primary', () => {
    const { container } = renderWithTheme(
      <Button variant="primary">Ver Cases</Button>
    )
    expect(container).toMatchSnapshot()
    expect(screen.getByRole('button', { name: /ver cases/i })).toHaveStyleRule(
      'background',
      '#FBD968'
    )
  })

  it('should render a button and with variant primary on hover', () => {
    renderWithTheme(<Button variant="primary">Ver Cases</Button>)

    expect(screen.getByRole('button', { name: /ver cases/i })).toHaveStyleRule(
      'background',
      '#F8D048',
      { modifier: ':hover' }
    )
  })
})
