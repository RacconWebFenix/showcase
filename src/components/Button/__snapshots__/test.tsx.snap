// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a button and with variant primary 1`] = `
.c0 {
  background: #FBD968;
  padding: 1.6rem 3.2rem;
  border: 0;
  cursor: pointer;
  color: #333333;
  font-weight: 500;
  font-size: 1.8rem;
  text-decoration: none;
  line-height: 2.0rem;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  transition: all ease 0.2s;
}

.c0:hover {
  background: #F8D048;
  box-shadow: 10px 10px 5px 0px #000000;
}

.c0:active {
  box-shadow: 5px 5px 10px 1px #000000;
}

<div>
  <button
    class="c0"
    variant="primary"
  >
    Ver Cases
  </button>
</div>
`;
